### PR TITLE
debezium/dbz#2102 Handle MySQL ENUM and SET literal escaping

### PR DESCRIPTION
**Pull request by minleejae**
Fixes debezium/dbz#2102

## Description
This change improves MySQL and MariaDB ENUM/SET allowed value handling by keeping parser output as parsed option values and moving schema-parameter escaping to the common enumerated-values helper.

It also updates MySQL JDBC sink type rendering tests to cover comma, quote, backslash, trailing backslash, and empty ENUM labels.

## Testing
- `git diff --check`
- `JAVA_HOME=/opt/homebrew/opt/openjdk/libexec/openjdk.jdk/Contents/Home ./mvnw -pl debezium-connector-common test -Dtest=io.debezium.data.EnumeratedValuesTest`

### CODE REVIEW

**Comment by jpechane** [comment]
@minleejae I know this is fix for source connector issue but could you please introduce a test on the sink side too to make sure it works end to end?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlPtAntlrDdlParser.java [hunk 1]
```diff
@@ -432,10 +432,9 @@ public static List<String> extractEnumAndSetOptions(List<String> enumValues) {
     }
 
     public static String escapeOption(String option) {
```
Comment (Naros) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlPtAntlrDdlParser.java]: I wonder if this should be named `unescapeOption`? The same applies to the other classes.

**Review by Naros — COMMENTED** [review]
Overall this LGTM, I've left just one nit-pick below. Feel free to follow-up on this separately if you'd prefer.

**Review by Naros — APPROVED** [review]

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlPtAntlrDdlParser.java (hunk 1 above)
Comment (minleejae) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/MySqlPtAntlrDdlParser.java]: Applied in [36b582a](https://github.com/debezium/debezium/pull/7565/commits/36b582ae92293fa790a36361cb73462b84b4774d)

**Review by Naros — APPROVED** [review]

**Comment by Naros** [comment]
Applied, thanks @minleejae :tada:
